### PR TITLE
Validate interface IP address; increase portchannel ID maximum to 9999

### PR DESCRIPTION
**Pull request by spenugondaa**
Validate interface IP address to reject unspecified/multicast/loopback/link-local-unicast addresses.

Increase maximum port channel ID to be 9999 instead of 256

~~~text
https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml
http://www.cidr-report.org/as2.0/reserved-ipv6.html

sonic(conf-if-Ethernet4)# ip address 0.1.1.1/24
%Error: Invalid IPv4 address 0.1.1.1
sonic(conf-if-Ethernet4)# ip address 255.1.1.1/24
%Error: Invalid IPv4 address 255.1.1.1
sonic(conf-if-Ethernet4)# ip address 0.0.0.0/24
%Error: Invalid IPv4 address 0.0.0.0
sonic(conf-if-Ethernet4)# ip address 127.0.0.1/24
%Error: Invalid IPv4 address 127.0.0.1
sonic(conf-if-Ethernet4)# ip address 224.0.0.1/24
%Error: Invalid IPv4 address 224.0.0.1
sonic(conf-if-Ethernet4)# ip address 169.254.0.1/24
%Error: Invalid IPv4 address 169.254.0.1

sonic(conf-if-Ethernet4)# ipv6 address ::1/128
%Error: Invalid IPv6 address ::1
sonic(conf-if-Ethernet4)# ipv6 address ::0/128
%Error: Invalid IPv6 address ::0
sonic(conf-if-Ethernet4)# ipv6 address FF00::/8
%Error: Invalid IPv6 address ff00::
sonic(conf-if-Ethernet4)# ipv6 address FE80::/10
%Error: Invalid IPv6 address FE80::
~~~

For portchannel ID:
~~~text
sonic(config)# interface PortChannel
    PortChannel identifier (1..9999)

sonic(config)# interface PortChannel 9876
sonic(conf-if-po9876)# mtu 9215
sonic(conf-if-po9876)# end
sonic# show interface PortChannel
:
PortChannel9876 is down, line protocol is down, mode LACP
Minimum number of links to bring PortChannel up is 1
Fallback: Disabled
MTU 9215

sonic# show PortChannel summary
Flags(oper-status):  D - Down U - Up
----------------------------------------------------------------------------------------------------------------------------
Group               PortChannel                   Type                Protocol       Member Ports
----------------------------------------------------------------------------------------------------------------------------
4                   PortChannel4   (D)            Eth                 LACP
8                   PortChannel8   (D)            Eth                 LACP
9876                PortChannel9876(D)            Eth                 LACP
sonic#

~~~

### CODE REVIEW

File: src/CLI/clitree/cli-xml/sonic_types.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@ limitations under the License.
     <PTYPE
         name="LAG_ID"
         method="integer"
-        pattern="1..256"
+        pattern="1..9999"
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/sonic_types.xml:56]: We cant just make this change without ensuring that every command that prints the interface name in the output works correctly, especially the ones that have column output.

Comment (spenugondaa) [reply src/CLI/clitree/cli-xml/sonic_types.xml:56]: Hi @joyas-joseph I did try "show interface PortChannel" and "show interface PortChannel 9876". Both are displaying non-table format.
The 'show interface status', 'show interface counters' are displaying data in tabular format however, they are displaying ethernet interfaces.
Other than this, I dont see any other command that is displaying name of an interface that is port channel.. 
The minimum width for the interface name everywhere seem to be 15 which is sufficient for the maximum "PortChannel9999"

Comment (spenugondaa) [reply src/CLI/clitree/cli-xml/sonic_types.xml:56]: Changed two renderers

Comment (bhavini-gada) [reply src/CLI/clitree/cli-xml/sonic_types.xml:56]: Please check show portchannel summary output as well.

**Pull request closed without merging**